lib/phoenix/endpoint/supervisor.ex

```diff
@@ -302,7 +302,7 @@ defmodule Phoenix.Endpoint.Supervisor do
     host   = host_to_binary(url[:host] || "localhost")
     port   = port_to_integer(url[:port] || port)
 
-    if host =~ ":" do
+    if host =~ "/" do
```

Comment:
WDYT about this then?

```suggestion
    if host =~ ~r"[^:]:\d" do
```